state/dbstore.go

```diff
@@ -107,14 +107,25 @@ func (s *DBStore) Delete(key string) (err error) {
 	return s.db.Delete([]byte(key), nil)
 }
 
-// Iterate entries which has a specific prefix
-// The key and value may be modified in the iterFunc
-func (s *DBStore) Iterate(prefix string, iterFunc func([]byte, []byte)) (err error) {
+// iterFunction is a function called on every key/value pair obtained
+// If true is returned in the stop variable, iteration will
+// stop, and by returning the error, that error will be
+// propagated to the called iterator method on Iterate.
+type iterFunction func([]byte, []byte) (stop bool, err error)
```

Comment:
i recommend named variables here like `(key, value []byte)`, easier to interpret

Reply:
This functionality is added also in https://github.com/ethersphere/swarm/pull/1554/files#r313813885

Reply:
Done.

This function was pulled from there only.